torch/_subclasses/fake_tensor.py

```diff
@@ -557,6 +557,10 @@ class FakeTensor(torch.Tensor):
     # this is an "infra" mode with lower dispatching precedence.
     _mode_key = torch._C._TorchDispatchModeKey.FAKE
 
+    # This is set when an tensor input is fakified. It is used when constructing
+    # a NestedIntSource from an offsets Tensor.
+    source = None
```

Comment:
It is set for every tensor actually! Which is not great, but hard to avoid because during fakification, we don't know yet whether this tensor will be used in NT construction yet.